File: programmers/Level1/solution/_48_핸드폰_번호_가리기/Solution.java
package programmers.Level1.solution._48_핸드폰_번호_가리기;

/**
 * https://school.programmers.co.kr/learn/courses/30/lessons/12948?language=java
 */
public class Solution {

    public static void main(String[] args) {

        String phoneNumber = "01033334444";

        Solution solution = new Solution();
        String result = solution.solution(phoneNumber);
        System.out.println("result: " + result);
    }

    public String solution(String phone_number) {

        StringBuilder answer = new StringBuilder();
        for (int i = 0; i < phone_number.length(); i++) {
            if (i < phone_number.length() - 4) {
                answer.append("*");
            } else {
                answer.append(phone_number.charAt(i));
            }
        }

        return answer.toString();
    }

}

File: programmers/Level1/solution/_48_핸드폰_번호_가리기/Solution.kt
package programmers.Level1.solution._48_핸드_번호_가리기

fun main() {
    var phone_number: String = "01032313231"
    var result = solution(phone_number)
    print("result: " + result)
}

fun solution(phone_number: String): String {
    var answer = StringBuilder()
    
    for (i in phone_number.indices) {
        if (i < phone_number.length - 4) {
            answer.append('*')
        } else {
            answer.append(phone_number.get(i))
        }
    }

    return answer.toString()
}
